lib/widget/switch.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_neumorphic/NeumorphicBoxShape.dart';
+import 'package:flutter_neumorphic/flutter_neumorphic.dart';
+
+// Constants below are for easy scaling tests.
+// Can be removed later.
+const double SCALE = 1.0;
+const double TRACK_WIDTH = 90 * SCALE;
+const double TRACK_HEIGHT = 45 * SCALE;
+const double THUMB_WIDTH = 31.5 * SCALE;
+const double THUMB_HEIGHT = 31.5 * SCALE;
+const double THUMB_CONTAINER_WIDTH = 90 * SCALE;
+const double THUMB_CONTAINER_HEIGHT = 36 * SCALE;
+
+class NeumorphicSwitchStyle {
+  final double depth;
+  final Color activeTrackColor;
+  final Color inactiveTrackColor;
+  final Color activeThumbColor;
+  final Color inactiveThumbColor;
+
+  const NeumorphicSwitchStyle(
+      {this.depth = -3,
+      this.activeTrackColor = NeumorphicColors.accent,
+      this.inactiveTrackColor = NeumorphicColors.background,
+      this.activeThumbColor = NeumorphicColors.background,
+      this.inactiveThumbColor = NeumorphicColors.background});
+}
+
+class NeumorphicSwitch extends StatefulWidget {
+  final bool value;
+  final ValueChanged<bool> onChanged;
+  final NeumorphicSwitchStyle style;
+
+  const NeumorphicSwitch({
+    this.style,
+    Key key,
+    this.value = false,
+    this.onChanged,
+  }) : super(key: key);
+
+  @override
+  _NeumorphicSwitchState createState() => _NeumorphicSwitchState();
+}
+
+class _NeumorphicSwitchState extends State<NeumorphicSwitch>
+    with SingleTickerProviderStateMixin {
+  Animation<Alignment> animation;
+  AnimationController controller;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = AnimationController(
+        duration: const Duration(milliseconds: 200), vsync: this);
+    animation = Tween<Alignment>(
+            begin: Alignment.centerLeft, end: Alignment.centerRight)
+        .animate(controller);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final NeumorphicTheme theme =
+        NeumorphicThemeProvider.findNeumorphicTheme(context);
+    final trackColor = widget.value == false
+        ? widget.style.inactiveTrackColor ?? theme.baseColor
+        : widget.style.activeTrackColor ?? theme.accentColor;
+    return SizedBox(
+      height: TRACK_HEIGHT,
+      width: TRACK_WIDTH,
+      child: Neumorphic(
+        shape: NeumorphicBoxShape.stadium(),
+        style: NeumorphicStyle(
+            depth: widget.style.depth * SCALE,
+            shape: NeumorphicShape.emboss,
+            baseColor: trackColor),
+        child: ClipRRect(
```

Comment:
you can remove this ClipRrect I think

it can clip the shadow and IMO is useless